### PR TITLE
[MM-43222]: render suggestion list dividers the correct way

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
#8976 introduced a regression with suggestion list dividers.
This PR should fix the regression and also creates a better way to selectively render only the dividers that are meant to be.

As a sidenote: we should look into making the divider types either an `enum` or a `union type` once we start migrating the `suggestion` components to typescript. (there already is a HW ticket available: https://github.com/mattermost/mattermost-server/issues/17278) This would make the type checking and props definition a bit easier.

#### Ticket Link
[MM-43222](https://mattermost.atlassian.net/browse/MM-43222)

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
fixes a regression with dividers in suggestion lists
```

### CODE REVIEW

**Review by matthewbirtch — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Looks good to me.

Looking back, I wish I'd made the dividers just list items with some special behaviour that made them unselectable. Even the original version that just had dividers in one place (I think it was the search box?) felt like a hack, and I feel like we've had to keep adding to that over time as those lists are changed. We'll have to remember that whenever we get around to redoing or replacing those components

**Comment by jgilliam17** [comment]
Thanks @michelengelen 
Tested manually, looks good. Can you please fix the failed circleci tests so I can run E2Es? 
- Verified suggestion list dividers for in channel members, special mentions and not in channel - as expected.

**Comment by michelengelen** [comment]
> Looks good to me.
> 
> Looking back, I wish I'd made the dividers just list items with some special behaviour that made them unselectable. Even the original version that just had dividers in one place (I think it was the search box?) felt like a hack, and I feel like we've had to keep adding to that over time as those lists are changed. We'll have to remember that whenever we get around to redoing or replacing those components

that might be a good candidate for a refactor. It's always a bit strange (and hard to follow) when the components get rendered due to some form of selection and it is not clear which component gets selected. In this case it took me good while to follow the code and find the culprit. :D

**Review by AshishDhama — APPROVED** [review]
LGTM !!!

**Comment by jgilliam17** [comment]
e2es running [here](https://automation-dashboard.vercel.app/cycles/544f3184-e1f7-4847-bb94-410f6130de56)

**Comment by jgilliam17** [comment]
@michelengelen Actually [report](https://mattermost-cypress-report.s3.amazonaws.com/187671-e7df999-onprem-ent-webapp-pr-10224-mm-ee-test:e7df999/mochawesome.html) shows 3 channel autocomplete failures

**Comment by michelengelen** [comment]
@jgilliam17 I have fixed the related fails. Sry for the extra round. 🙇🏼

**Comment by jgilliam17** [comment]
Thanks @michelengelen the tests have a different failure, not able to find  `MY CHANNELS` divider text. Latest report [here](https://mattermost-cypress-report.s3.amazonaws.com/187753-cb6e740-onprem-ent-webapp-pr-10224-mattermost/mm-ee-testcb6e740/mochawesome.html)

**Comment by jgilliam17** [comment]
Thanks @michelengelen 
Tested, looks good to merge. Latest [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/187928-4db94d3-onprem-ent-webapp-pr-10224-mattermost/mm-ee-test4db94d3/mochawesome.html) doesn't have any PR related failures.

**Review by jgilliam17 — APPROVED** [review]

**Comment by jgilliam17** [comment]
@amyblais Do you think we should include this into v6.7? Let me know - thanks 🙂

**Comment by amyblais** [comment]
It doesn't seem to be a regression, so not necessarily urgent for v6.7.
